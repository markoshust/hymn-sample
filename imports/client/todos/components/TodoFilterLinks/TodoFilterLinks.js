import React from 'react';
import TodoFilterLink from '../../containers/TodoFilterLink';

const TodoFilterLinks = () => (
  <p>
    Show:
    {" "}
    <TodoFilterLink filter="SHOW_ALL">
      All
    </TodoFilterLink>
    {", "}
    <TodoFilterLink filter="SHOW_ACTIVE">
      Active
    </TodoFilterLink>
    {", "}
    <TodoFilterLink filter="SHOW_COMPLETED">
      Completed
    </TodoFilterLink>
  </p>
);

export default TodoFilterLinks;
