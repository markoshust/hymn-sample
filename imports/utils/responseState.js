const responseState = (response, ...state) => (
  response[state] ? response[state]() : state
);

export default responseState;
