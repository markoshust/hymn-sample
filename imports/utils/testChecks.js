import renderer from 'react-test-renderer';

const componentCheck = (component) => {
  const tree = component.toJSON();

  expect(tree).toMatchSnapshot();

  return {
    component,
    tree,
  };
};

const renderCheck = (children) => {
  const component = renderer.create(children);

  return componentCheck(component);
};

export {
  componentCheck,
  renderCheck,
};
