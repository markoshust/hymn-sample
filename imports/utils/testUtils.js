import renderer from 'react-test-renderer';

const snap = (children) => {
  const component = renderer.create(children);
  const tree = component.toJSON();

  expect(tree).toMatchSnapshot();

  return {
    component,
    tree,
  };
};

export {
  snap, // eslint-disable-line import/prefer-default-export
};
