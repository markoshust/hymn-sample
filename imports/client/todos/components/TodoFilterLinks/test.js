jest.mock('../../containers/TodoFilterLink');

/* eslint-disable import/imports-first */
import React from 'react';
import { renderCheck } from 'utils/testChecks';
import TodoFilterLinks from './TodoFilterLinks';
/* eslint-enable import/imports-first */

describe('<TodoFilterLinks />', () => {
  it('renders correctly', () => {
    renderCheck(
      <TodoFilterLinks />
    );
  });
});
