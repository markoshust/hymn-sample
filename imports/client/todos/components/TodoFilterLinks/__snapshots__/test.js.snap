exports[`<TodoFilterLinks /> renders correctly 1`] = `
<p>
  Show:
   
  , 
  , 
</p>
`;
